enhancements/network/ovn-kubernetes-localnet-api.md

```diff
@@ -0,0 +1,566 @@
+    // +optional
+    Localnet *LocalnetConfig `json:"localnet,omitempty"`
+}
+```
+
+The CUDN spec should have additional validation rule for `spec.network.topology` field:
+```go
+// ClusterUserDefinedNetworkSpec defines the desired state of ClusterUserDefinedNetwork.
+type ClusterUserDefinedNetworkSpec struct {
+    ...
+    // +required
+    Network NetworkSpec `json:"network"`
+}
+```
+
+As of today the CUDN has validation to ensure `spec.network` is immutable:
+```go
+// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Network spec is immutable"
+```
+This validation should be changed in a way it allows mutating localnet topology configurations, 
+at least MTU, VLAN, physicalNetworkName and excludeSubnets. 
+The proposed change is to invert the immutable-spec validation on CUDN `spec.network` as follows:
+1. Remove mutation restrictions on `spec.network`.
+2. Add CEL rule validations to restrict mutations of:
+   `spec.network.topology`, `spec.network.layer2`, and `spec.network.layer3`.
+3. Add CEL rule validations to restrict mutation of localnet topology settings: 
+   `spec.network.localnet.role`, `spec.network.localnet.subnets` and `spec.network.topology.ipam`.
+
+Localnet topology configuration type 
+```go
+// +kubebuilder:validation:XValidation:rule="has(self.ipam) && has(self.ipam.mode) && self.ipam.mode != 'Enabled' || has(self.subnets)", message="Subnets is required with ipam.mode is Enabled or unset"
+// +kubebuilder:validation:XValidation:rule="!has(self.ipam) || !has(self.ipam.mode) || self.ipam.mode != 'Disabled' || !has(self.subnets)", message="Subnets must be unset when ipam.mode is Disabled"
+// +kubebuilder:validation:XValidation:rule="!has(self.excludeSubnets) || has(self.excludeSubnets) && has(self.subnets)", message="excludeSubnets must be unset when subents is unset"
+// +kubebuilder:validation:XValidation:rule="!has(self.subnets) || !has(self.mtu) || !self.subnets.exists_one(i, isCIDR(i) && cidr(i).ip().family() == 6) || self.mtu >= 1280", message="MTU should be greater than or equal to 1280 when IPv6 subent is used"
+type LocalnetConfig struct {
+    // role describes the network role in the pod, required.
+    // Whether the pod interface will act as primary or secondary.
+    // For Localnet topology only `Secondary` is allowed.
+    // Secondary network is only assigned to pods that use `k8s.v1.cni.cncf.io/networks` annotation to select given network.
+    // +kubebuilder:validation:Enum=Secondary
+    // +required
+    Role NetworkRole `json:"role"`
+    
+    // physicalNetworkName points to the OVS bridge-mapping's network-name configured in the nodes, required.
+    // In case OVS bridge-mapping is defined by Kubernetes-nmstate with `NodeNetworkConfigurationPolicy` (NNCP),
+    // this field should point to the value of the NNCP spec.desiredState.ovn.bridge-mappings.
+    // Max length is 253, cannot contain `,` or `:` characters.
+    // +kubebuilder:validation:MinLength=1
```

Comment:
Include in the godoc that the minimum length is 1 (i.e the empty string `""` is not allowed)

Reply:
Done